source/Engine.cpp

```diff
@@ -1461,10 +1461,26 @@ void Engine::CalculateStep()
 
 	// Keep track of the flagship to see if it jumps or enters a wormhole this turn.
 	const Ship *flagship = player.Flagship();
+	bool flagshipWasUntargetable = (flagship && !flagship->IsTargetable());
 	bool wasHyperspacing = (flagship && flagship->IsEnteringHyperspace());
-	// Move all the ships.
+	const System *flagshipSystem = (flagship ? flagship->GetSystem() : nullptr);
+	// First, move the player's flagship.
+	MoveShip(player.FlagshipPtr());
+	bool flagshipIsUntargetable = (flagship && !flagship->IsTargetable());
+	bool flagshipBecameTargetable = flagshipWasUntargetable && !flagshipIsUntargetable;
```

Comment:
We're only ever using the negation of this first boolean, so just change it to is targetable rather than keeping it as is untargetable.
```suggestion
	bool flagshipIsTargetable = (flagship && flagship->IsTargetable());
	bool flagshipBecameTargetable = flagshipWasUntargetable && flagshipIsTargetable;
```